src/main/java/se/liu/ida/hefquin/engine/queryplan/utils/TextBasedPhysicalPlanPrinterImpl2.java

```diff
@@ -0,0 +1,354 @@
+		out.append( indentLevelString + "hashRJoin (" + op.getID() + ") " );
+		printLogicalOpForPhysicalOp( op.getLogicalOperator(), out, indentLevelString, indentLevelStringForOpDetail );
+	}
+	
+	protected void printOperatorInfoForOpIndexNestedLoopsJoin ( final PhysicalOpIndexNestedLoopsJoin op, final PrintStream out, final String indentLevelString, final String indentLevelStringForOpDetail ) {
+		out.append( indentLevelString + "hashRJoin (" + op.getID() + ") " );
+		printLogicalOpForPhysicalOp( op.getLogicalOperator(), out, indentLevelString, indentLevelStringForOpDetail );
+	}
+
+	protected void printOperatorInfoForOpLocalToGlobal ( final PhysicalOpLocalToGlobal op, final PrintStream out, final String indentLevelString, final String indentLevelStringForOpDetail ) {
+		out.append( indentLevelString + "l2g (" + op.getID() + ") " );
+		printLogicalOpForPhysicalOp( op.getLogicalOperator(), out, indentLevelString, indentLevelStringForOpDetail );
+	}
+	
+	protected void printOperatorInfoForMultiwayUnion ( final PhysicalOpMultiwayUnion op, final PrintStream out, final String indentLevelString, final String indentLevelStringForOpDetail ) {
+		out.append( indentLevelString + "multiwayUnion (" + op.getID() + ") " );
+		printLogicalOpForPhysicalOp( op.getLogicalOperator(), out, indentLevelString, indentLevelStringForOpDetail );
+	}
+
+	protected void printOperatorInfoForOpNaiveNestedLoopsJoin ( final PhysicalOpNaiveNestedLoopsJoin op, final PrintStream out, final String indentLevelString, final String indentLevelStringForOpDetail ) {
+		out.append( indentLevelString + "naiveNestedLoop (" + op.getID() + ") " );
+		printLogicalOpForPhysicalOp( op.getLogicalOperator(), out, indentLevelString, indentLevelStringForOpDetail );
+	}
+
+	protected void printOperatorInfoForOpParallelMultiLeftJoin ( final PhysicalOpParallelMultiLeftJoin op, final PrintStream out, final String indentLevelString, final String indentLevelStringForOpDetail ) {
+		out.append( indentLevelString + "parallelMultiLeftJoin (" + op.getID() + ") " );
+		out.append( System.lineSeparator() );
+		//printLogicalOpForPhysicalOp( op.getLogicalOperator(), out, indentLevelString, indentLevelStringForOpDetail );
+	}
+	
+	protected void printOperatorInfoForRequest( final PhysicalOpRequest op, final PrintStream out, final String indentLevelString, final String indentLevelStringForOpDetail ) {
+		final LogicalOpRequest lop = op.getLogicalOperator();
+		final DataRetrievalRequest req = lop.getRequest();
+		out.append( indentLevelString + "req (" + lop.getID() + ")" );
+		out.append( System.lineSeparator() );
+		printFederationMember( lop.getFederationMember(), indentLevelStringForOpDetail, out );
+		out.append( indentLevelStringForOpDetail + "  - pattern (" + req.hashCode() +  ") (" + req.toString() + ")" );
+		out.append( System.lineSeparator() );
+		out.append( indentLevelStringForOpDetail );
+		out.append( System.lineSeparator() );
```

Comment:
Why these?

And why is there now `printLogicalOpForPhysicalOp` in this function?

Reply:
These two lines are for giving an extra space line after "-pattern".